website/content/contribute/module/index.md

```diff
@@ -0,0 +1,199 @@
+---
+title: Contributing a Module
+date: 2020-11-28T22:10:51+02:00
+draft: false
+weight: 1
+---
+
+This page describes what exactly must be provided when contributing a module, and how to make the control plane aware of the new module so that it can be used.
+
+
+
+Steps for creating a module:
+1. Implement the logic of the module you are contributing - either directly in the [Module Workload](#module-workload) or in an external component.  If the logic is in an external component, then the module workload should act as a client - i.e. receiving paramaters from the control plane and passing them to the external component.
+2. [Create the Module Helm Chart](#module-helm-chart) that will be used by the control plane to deploy the module workload, update it, and delete it as necessary.
+3. [Create the M4DModule YAML](#m4dmodule-yaml) which describes the capabilities of the module workload, in which flows it should be considered for inclusion, its supported interfaces, and the link to the module helm chart.
+4. [Register the Module](#register-the-module) to make it available to the control plane.
+
+These steps are described in the following sections in more detail, so that you can create your own modules for use by the Mesh for Data.  Note that a new module is maintained in its own git repository, separate from the core [Mesh for Data repository] (https://github.ibm.com/data-mesh-research/datamesh).
+
+
+## Module Workload
+
+The module workload is associated with a specific user workload and is deployed by the control plane.  It may implement the logic required itself, or it may be a client interface to an external component.  
+
+In either case its input parameters must match the relevant type of `ModuleArguments`.  The details of the different types of arguments are defined in [blueprint_types.go](/Users/sima/Work/Dev/go/sima-m4d/the-mesh-for-data/manager/apis/app/v1alpha1/blueprint_types.go). 
+
+```
+type ModuleArguments struct {
+	// CopyArgs are parameters specific to modules that do implicit copies of data from one data store to another.
+	// +optional
+	Copy *CopyModuleArgs `json:"copy,omitempty"`
+
+	// ReadArgs are parameters that are specific to modules that enable user workloads to read data or are involved in preparing data to be read.
+	// +optional
+	Read []ReadModuleArgs `json:"read,omitempty"`
+
+	// WriteArgs are parameters that are specific to modules that enable an application to write data
+	// +optional
+	Write []WriteModuleArgs `json:"write,omitempty"`
+}
+```
```

Comment:
We have this in the API reference docs: https://ibm.github.io/the-mesh-for-data/docs/reference/api/generated/app/#k8s-api-github-com-ibm-the-mesh-for-data-manager-apis-app-v1alpha1-modulearguments

Reply:
fixed